### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by franckevva**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  "license": "MIT",
+  "scripts": {
+    "ng": "ng",
+    "start": "ng serve",
+    "build": "ng build",
+    "lint": "ng lint"
+  },
+  "private": true,
+  "dependencies": {
+    "@angular/animations": "^5.0.0",
```
Comment (franckevva) [inline package.json]: use accurate version
remove `^` and `~`

File: src/app/app.component.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-root',
+  template: `<router-outlet></router-outlet>`,
+})
+export class AppComponent {
+  title = 'app';
```
Comment (franckevva) [inline src/app/app.component.ts]: remove unused variables

File: src/app/dashboard/line-chart/line-chart.component.ts [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+  private lineChart;
+  private timer;
+
+  public ngOnInit() {
+    this.options = {
+      container: d3.select('.line-chart__container'),
+      maxX: 14,
+      xStep: 0.125,
+      xDrawStep: 4,
+      rowBgColor: '#4a4a4a',
```
Comment (franckevva) [inline src/app/dashboard/line-chart/line-chart.component.ts]: can it make sense to put colors in variables?
are these colors still used somewhere?

File: src/app/dashboard/todo-list/todo-list.component.scss [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      color: $todo-input-text-label-color;
+      top: 0;
+      &:after {
+        background-color: $todo-input-text-highlight-color;
+        bottom: 0;
+      }
+    }
+  }
+  .mdl-card__actions {
+    padding: 8px 16px;
```
Comment (franckevva) [inline src/app/dashboard/todo-list/todo-list.component.scss]: try to use more `rem` than `px`

File: src/app/forms/employer-form/employer-form.component.scss [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+.employer-form {
+  padding: 0;
+  width: $employer-form-width;
+  background-color: $background-color;
+  margin: 16px auto;
+
+  .mdl-radio {
+    width: 20%;
+    margin-top: 22px;
```
Comment (franckevva) [inline src/app/forms/employer-form/employer-form.component.scss]: use `rem`

File: src/theme/scss/_mixins.scss [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+  line-height: 13px;
+  font-weight: 400;
+  color: white;
+}
+
+@mixin colored-button ($color) {
+  &.mdl-button--raised {
+    &, &:hover {
+      background-color: $color;
+      color: #ffffff;
```
Comment (franckevva) [inline src/theme/scss/_mixins.scss]: `$color-white`

File: src/theme/scss/badge/badge.scss [hunk 7]
```diff
@@ -0,0 +1,29 @@
+//@mixin colored-badge ($color) {
```
Comment (franckevva) [inline src/theme/scss/badge/badge.scss]: remove unused comments

File: src/theme/scss/button/button.scss [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+      &[disabled], &.mdl-button--raised {
+        i {
+          background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' style='width:24px;height:24px' viewBox='0 0 24 24'%3E%3Cpath fill='%23ffffff' d='M22.46,6C21.69,6.35 20.86,6.58 20,6.69C20.88,6.16 21.56,5.32 21.88,4.31C21.05,4.81 20.13,5.16 19.16,5.36C18.37,4.5 17.26,4 16,4C13.65,4 11.73,5.92 11.73,8.29C11.73,8.63 11.77,8.96 11.84,9.27C8.28,9.09 5.11,7.38 3,4.79C2.63,5.42 2.42,6.16 2.42,6.94C2.42,8.43 3.17,9.75 4.33,10.5C3.62,10.5 2.96,10.3 2.38,10C2.38,10 2.38,10 2.38,10.03C2.38,12.11 3.86,13.85 5.82,14.24C5.46,14.34 5.08,14.39 4.69,14.39C4.42,14.39 4.15,14.36 3.89,14.31C4.43,16 6,17.26 7.89,17.29C6.43,18.45 4.58,19.13 2.56,19.13C2.22,19.13 1.88,19.11 1.54,19.07C3.44,20.29 5.7,21 8.12,21C16,21 20.33,14.46 20.33,8.79C20.33,8.6 20.33,8.42 20.32,8.23C21.16,7.63 21.88,6.87 22.46,6Z' /%3E%3C/svg%3E");
+        }
+      }
+    }
+    &--linkedin {
+      &.mdl-button--raised:not([disabled]) {
+        &, &:hover {
+          background-color: #0083be;
```
Comment (franckevva) [inline src/theme/scss/button/button.scss]: try to set colors in variables
and don't forget `$color-white`

File: src/theme/scss/button/button.scss [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+
+      }
+      i {
+        background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' style='width:24px;height:24px' viewBox='0 0 24 24'%3E%3Cpath fill='%23ffffff' d='M12,2A10,10 0 0,0 2,12C2,16.42 4.87,20.17 8.84,21.5C9.34,21.58 9.5,21.27 9.5,21C9.5,20.77 9.5,20.14 9.5,19.31C6.73,19.91 6.14,17.97 6.14,17.97C5.68,16.81 5.03,16.5 5.03,16.5C4.12,15.88 5.1,15.9 5.1,15.9C6.1,15.97 6.63,16.93 6.63,16.93C7.5,18.45 8.97,18 9.54,17.76C9.63,17.11 9.89,16.67 10.17,16.42C7.95,16.17 5.62,15.31 5.62,11.5C5.62,10.39 6,9.5 6.65,8.79C6.55,8.54 6.2,7.5 6.75,6.15C6.75,6.15 7.59,5.88 9.5,7.17C10.29,6.95 11.15,6.84 12,6.84C12.85,6.84 13.71,6.95 14.5,7.17C16.41,5.88 17.25,6.15 17.25,6.15C17.8,7.5 17.45,8.54 17.35,8.79C18,9.5 18.38,10.39 18.38,11.5C18.38,15.32 16.04,16.16 13.81,16.41C14.17,16.72 14.5,17.33 14.5,18.26C14.5,19.6 14.5,20.68 14.5,21C14.5,21.27 14.66,21.59 15.17,21.5C19.14,20.16 22,16.42 22,12A10,10 0 0,0 12,2Z' /%3E%3C/svg%3E");
+      }
+    }
+    &--dribbble  {
+      &.mdl-button--raised:not([disabled]) {
+        &, &:hover {
+          background-color: #f26798;
```
Comment (franckevva) [inline src/theme/scss/button/button.scss]: try to set colors in variables
and don't forget `$color-white`

File: src/theme/scss/chip/chip.scss [hunk 10]
```diff
@@ -0,0 +1,6 @@
+.mdl-chip {
+  color: black;
+  &__action {
+    color: inherit;
+  }
+}
```
Comment (franckevva) [inline src/theme/scss/chip/chip.scss]: newline at the end of the file
add pre-commit rule  `lint` on the package.json

File: src/theme/scss/data-table/data-table.scss [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+    &:last-of-type {
+      padding-right: 16px;
+    }
+  }
+  tr:hover {
+    background-clip: padding-box;
+  }
+  &.bordered-table {
+    td, th {
+      border-right: 1px solid #606060;
```
Comment (franckevva) [inline src/theme/scss/data-table/data-table.scss]: colors to variables

File: src/theme/scss/menu/menu.scss [hunk 12]
```diff
@@ -0,0 +1,10 @@
+.mdl-menu {
+  width: $dropdown-menu-width;
+  .mdl-menu__item.mdl-list__item {
+    display: flex;
+  }
+}
+
+.mdl-menu__outline {
+  @include shadow-4dp();
+}
```
Comment (franckevva) [inline src/theme/scss/menu/menu.scss]: new line

File: src/theme/scss/palette.scss [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+  color: $color-purple !important;
+}
+
+.color--teal {
+  background-color: $color-teal !important;
+}
+
+.color-text--teal {
+  color: $color-teal !important;
+}
```
Comment (franckevva) [inline src/theme/scss/palette.scss]: new line

File: src/theme/widgets/charts/charts.scss [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+      }
+    }
+    .legend {
+      max-width: 140px;
+    }
+  }
+
+  .chart1__container .nv-label {
+    text {
+      fill: white !important;
```
Comment (franckevva) [inline src/theme/widgets/charts/charts.scss]: white? -> `$color-white`

File: src/theme/widgets/charts/charts.scss [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+    height: 100%;
+    svg {
+      height: 240px;
+      max-width: calc(100% - 140px);
+      .nv-bar {
+        text {
+          fill: #ffffff !important;
+          opacity: 0.9;
+        }
+      }
```
Comment (franckevva) [inline src/theme/widgets/charts/charts.scss]: `$color-white`

File: src/theme/widgets/map/map.scss [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+    background: $card-title-background-color;
+    border-radius: 50%;
+    width: 170px;
+    height: 170px;
+    margin: 0 10px 5px 15px;
+  }
+  .map__target-picture {
+    height: 100%;
+    transition: opacity 0.3s;
+    background: url("/assets/images/map_target_images_sprite.png") no-repeat top left;
```
Comment (franckevva) [inline src/theme/widgets/map/map.scss]: you have `$image_path`. Why don't you use it?

File: src/theme/widgets/ui-cards/ui-cards.scss [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+    }
+    &:nth-last-of-type(2) .mdl-card__title {
+      background-image: url("/assets/images/tokyo.jpg");
+    }
+    &:nth-last-of-type(3) .mdl-card__title {
+      background-image: url("/assets/images/istanbul.jpg");
+    }
+  }
+  small {
+    color: #aaaaaa;
```
Comment (franckevva) [inline src/theme/widgets/ui-cards/ui-cards.scss]: colors to variables

File: src/theme/widgets/ui-components/ui-components.scss [hunk 18]
```diff
@@ -0,0 +1,11 @@
+.ui-components {
+  .mdl-card {
+    height: 100%;
+  }
+  .mdl-list__item {
+    padding: 8px 16px;
```
Comment (franckevva) [inline src/theme/widgets/ui-components/ui-components.scss]: use `rem`, in the possible places

File: tslint.json [hunk 19]
```diff
@@ -0,0 +1,144 @@
+{
+  "rulesDirectory": [
+    "node_modules/codelyzer"
+  ],
+  "rules": {
```
Comment (franckevva) [inline tslint.json]: Do you really need so many rules? 
Use `"extends"` property and, for example, this [tslint-angular]( https://www.npmjs.com/package/tslint-angular) config

```
"rulesDirectory": [
    "node_modules/codelyzer"
  ],
  "extends": [
    "tslint-angular",
    "tslint:recommended"
  ],
  "rules": {
     // some custom rule
    // at first remove all and than add only you need change 
    }
}
```

File: package.json [hunk 20]
```diff
@@ -0,0 +1,57 @@
...
+    "dark material design",
+    "angular",
+    "ng"
+  ],
+  "license": "MIT",
+  "scripts": {
+    "ng": "ng",
+    "start": "ng serve",
+    "build": "ng build",
+    "lint": "ng lint"
```
Comment (franckevva) [inline package.json]: - recommend adding `"lint:fix": "ng lint --fix"` rule
- add [stylelint](https://github.com/stylelint/stylelint) for scss files

File: package.json [hunk 21]
```diff
@@ -0,0 +1,57 @@
...
+  },
+  "private": true,
```
Comment (franckevva) [inline package.json]: add [pre-commit](https://www.npmjs.com/package/pre-commit)
example using: 
`"pre-commit": [
    "lint"
  ]`
where `lint` it's a rule from script, that run lint and stylelint

File: src/app/app.component.ts (hunk 2 above)
Comment (ByPort) [reply src/app/app.component.ts]: ✔️

File: src/theme/scss/_mixins.scss (hunk 6 above)
Comment (ByPort) [reply src/theme/scss/_mixins.scss]: ✔️

File: src/theme/scss/badge/badge.scss (hunk 7 above)
Comment (ByPort) [reply src/theme/scss/badge/badge.scss]: ✔️

File: src/theme/scss/chip/chip.scss (hunk 10 above)
Comment (ByPort) [reply src/theme/scss/chip/chip.scss]: ✔️

File: src/theme/scss/menu/menu.scss (hunk 12 above)
Comment (ByPort) [reply src/theme/scss/menu/menu.scss]: ✔️

File: src/theme/scss/palette.scss (hunk 13 above)
Comment (ByPort) [reply src/theme/scss/palette.scss]: ✔️

File: src/theme/widgets/charts/charts.scss (hunk 14 above)
Comment (ByPort) [reply src/theme/widgets/charts/charts.scss]: ✔️

File: src/theme/widgets/charts/charts.scss (hunk 15 above)
Comment (ByPort) [reply src/theme/widgets/charts/charts.scss]: ✔️

File: src/theme/widgets/map/map.scss (hunk 16 above)
Comment (ByPort) [reply src/theme/widgets/map/map.scss]: ✔️

File: tslint.json (hunk 19 above)
Comment (ByPort) [reply tslint.json]: ✔️

File: package.json (hunk 20 above)
Comment (ByPort) [reply package.json]: ✔️

File: package.json (hunk 21 above)
Comment (ByPort) [reply package.json]: ✔️

File: src/app/ui/buttons/buttons.component.scss [hunk 22]
```diff
@@ -0,0 +1,32 @@
+.ui-buttons {
+  h6 {
+    margin-top: 8px;
+    margin-bottom: 0;
+  }
+  .mdl-button--icon .material-icons {
+    font-size: 20px;
+  }
+  .mdl-list__item {
+    padding: 8px 16px ;
```
Comment (franckevva) [inline src/app/ui/buttons/buttons.component.scss]: try use `rem`

File: src/app/ui/cards/cards.component.scss [hunk 23]
```diff
@@ -0,0 +1,47 @@
...
+    }
+    &:nth-last-of-type(2) .mdl-card__title {
+      background-image: url("/assets/images/tokyo.jpg");
+    }
+    &:nth-last-of-type(3) .mdl-card__title {
+      background-image: url("/assets/images/istanbul.jpg");
+    }
+  }
+  small {
+    color: #aaaaaa;
```
Comment (franckevva) [inline src/app/ui/cards/cards.component.scss]: move colors to variables

File: src/theme/components/base-card/base-card.module.ts [hunk 24]
```diff
@@ -0,0 +1,31 @@
...
+  BaseCardBodyComponent,
+  BaseCardActionsComponent,
+} from './base-card.component';
+
+@NgModule({
+  imports: [
+    CommonModule,
+  ],
+  declarations: [
+    BaseCardComponent,
```
Comment (franckevva) [inline src/theme/components/base-card/base-card.module.ts]: it's comfortable create `const COMPONENTS = [ ]` and than use it such
``` 
declarations: [...COMPONENTS],
exports: [...COMPONENTS],
```

File: src/app/dashboard/weather/weather.component.scss [hunk 25]
```diff
@@ -24,7 +24,7 @@
       @include typo-weather-description;
 
       position: relative;
-      color: $color-white;
+      color: unquote("rgb(#{$color-white})");
```
Comment (franckevva) [inline src/app/dashboard/weather/weather.component.scss]: `color: $color-white;` 
It doesn't work? Why you need unquote and rgb?

**Pull request closed without merging**